demos/nonlinear_QG_winddrivengyre/qg_winddrivengyre.py.rst

```diff
@@ -0,0 +1,279 @@
+rotational Froude number.
+
+The non-conservative aspects of this model
+occur because of :math:`r`, the strength of the bottom drag, and
+:math:`Q_{\textrm{winds}}`, the vorticity of the winds. We pick the wind forcing
+as to generate a single gyre,
+
+.. math:: Q_{\textrm{winds}} = \tau \cos\left( \pi \left[\frac{y}{L_y} - \frac{1}{2} \right] \right)
+
+where :math:`L_y` is the length of our domain and :math:`\tau` is the strength of our wind forcing. By putting a :math:`2` in front of the :math:`\pi` we
+get a double gyre :cite:`Vallis:2006`.
+
+If we only look for steady solutions
+in time, we can ignore the time derivative term, and we get
+
+.. math::
+
+   \begin{gathered}
+   (\vec{u} \cdot \vec\nabla)\left( \nabla^2 \psi - F \psi\right)
+   + \beta \frac{\partial \psi}{\partial x} = - rq + Q_{\textrm{winds}} 
+   \end{gathered}
+
+We can write this out in one equation and is the nonlinear Stommel
+problem.
+
+.. math::
+
+   \begin{gathered}
+   \vec u \cdot \vec\nabla \left( \nabla^2 \psi \right) + r(\nabla^{2} \psi - F\psi) + \beta \frac{\partial \psi}{\partial x} =  Q_{\textrm{winds}} 
+   \end{gathered}
+
+Note that we dropped the :math:`-F \psi` term in the nonlinear advection
+because the streamfunction does not change following the flow, and
+therefore, we can neglect that term entirely.
+
+Weak Formulation
+================
+
+To build the weak form of the problem in Firedrake we must find the weak
+form of this equation. We begin by multiplying this equation by a Test
+Function, :math:`\phi`, which is in the same space as the
+streamfunction, and then integrate over the domain :math:`\Omega`,
+
+.. math:: \int_{\Omega} \phi (\vec u \cdot \vec\nabla) \nabla^2 \psi \,\mathrm{d}x  +  r\phi (\nabla^{2} \psi - F\psi)\,\mathrm{d}x + \beta\phi\frac{\partial \psi}{\partial x} \,\mathrm{d}x =  \int_{\Omega} \phi \cdot Q_{\textrm{winds}} \,\mathrm{d}x
+
+The nonlinear term can be rewritten using the fact that the velocity is
+divergent free and then integrating by parts,
+
+.. math::
+
+   \begin{aligned}
+   \int_{\Omega} \phi (\vec u \cdot \vec\nabla) \nabla^2 \psi
+   =  \int_{\Omega} \phi \vec\nabla \cdot \left(\vec u (\nabla^2 \psi)\right) 
+   = - \int_{\Omega}( \vec\nabla \phi \cdot \vec u){\nabla}^{2}\psi \, \mathrm{d}x.\end{aligned}
+
+Note that because we have no normal flow boundary conditions the
+boundary contribution is zero. For the term with bottom drag we
+integrate by parts and use the fact that the streamfunction is zero on
+the walls
+
+.. math::
+
+   \begin{aligned}
+   \int_{\Omega} r \phi \left( \vec{\nabla}^2 \psi - F \psi \right) \, \mathrm{d}x & 
+   = -r \int_{\Omega}  \Big(\vec{\nabla}\phi \cdot \vec{\nabla}\psi
+   + F \phi \psi \Big)\, \mathrm{d}x
+   + r \int_{\partial\Omega} \phi \cdot \frac{\partial \psi}{\partial n} \,\mathrm{d}s
+  \end{aligned}
+
+The boundary integral above banishes because are are setting the streamfunction to be zero on the boundary.
```

Comment:
spelling: "vanishes", not "banishes" (although good typo!)
also, repeated "are" (should be "we are").